rfc-106-docker-for-local-development.md

```diff
@@ -0,0 +1,41 @@
+# RFC 106: Use Docker for local development
+
+## Summary
+
+Adopt a Docker-based approach for local development instead of the Vagrant VM.
+
+## Problem
+
+We currently rely on the Vagrant development VM for the day to day dev cycle.
+
+- It has [many documented issues](https://docs.publishing.service.gov.uk/manual.html#development-vm)
+- Changes to puppet often result in a broken VM, because the infrastructure team don't use it.
+- We are adopting more cloud-based services into our stack, like S3 and Amazon's Elasticsearch. This means that our development environment doesn't look like production anymore, but we still need code in govuk-puppet.
```

Comment:
We also have that problem in CI.  Will we be dockerising CI too?

Reply:
Very good point. I guess the answer is yes!

Reply:
Containerising will happen as part of moving to Concourse CI

Reply:
It will, but until that happens we'll still have this problem of needing the puppet code to manage services in a different way to production.

Reply:
On the subject of concourse I'd like to point you to the work that has been going on with the [GDS Supported Platform](https://github.com/alphagov/gsp). 

The platform provides CICD, Docker Registry and Notary tools in the cluster. 
https://github.com/alphagov/gsp/blob/master/docs/architecture/gsp-architecture-overview.md

Its also worth making you aware of the [gsp-local](https://github.com/alphagov/gsp/blob/0b30ca7d05637f4660b67b9df8a0c3581caa37ec/docs/gds-supported-platform/getting-started-gsp-local.md) work which allows you to provision a local instance of the gsp on docker complete with CICD/Registry.  This has recently undergone a round of user research and we are in the process of tweaking the documentation that describes how to use it.